File: _stats_build/statistics/wr_ao2r.rb
require_relative "abstract/ao_rounds"

class WrAo2r < AoRounds
  def initialize
    @title = "World record Ao2R (Average of 2 Rounds) history"
    @title_zh = "世界纪录 Ao2R（双轮平均）历史"
    @note = "Ao2R: average of the 2 round averages (R1 + Final) in a competition."
    @note_zh = "Ao2R：一场比赛中两轮平均成绩的均值（第一轮 + 决赛）。"
    @table_header = { "Result" => :right, "Improvement" => :right, "Days" => :right, "Person" => :left, "Competition" => :left, "Date" => :left, "Details" => :left }
  end

  def round_count
    2
  end
end

File: core/jobs/stats-build/src/statistics/wr_ao2r.ts
// NOTE: Ao2R——一场比赛恰好 2 轮时各轮 average 的均值
import { AoRounds } from '../core/ao_rounds.js';

export class WrAo2r extends AoRounds {
  constructor() {
    super();
    this.title = 'Ao2R';
    this.titleZh = 'Ao2R';
    this.note = 'Average of 2 Rounds: mean of averages when a person competed in exactly 2 rounds at a competition.';
    this.noteZh = '两轮平均：一场比赛中恰好参加 2 轮时各轮平均的均值。';
  }
  roundCount() { return 2; }
}
